tests/integration/piv_worker_test.py

```diff
@@ -147,3 +149,23 @@ def test_mq_message_received_and_stored(
         # Check that MQ message is received and stored in DB
         mq_handler.publish(str('{"key": "Some valid JSON"}'), PIV_CONTRIBUTOR_ID)
         wait_until(lambda: RealTimeUpdate.query.count() == 1)
+
+
+@patch("kirin.command.piv_worker.PIV_WORKER_REDIS_TIMEOUT_LOCK", timedelta(seconds=0))
+def test_redis_lock_update(
+    test_client,
+    pg_docker_fixture,
+    rabbitmq_docker_fixture,
+    broker_connection,
+    mq_handler,
+):
+    logger = logging.getLogger("kirin.command.piv_worker")
+    with patch.object(logger, "debug") as mock_debug:
+        with PivWorkerTest(test_client, rabbitmq_docker_fixture.url, broker_connection, pg_docker_fixture):
+            # Check that PivWorker is creating the queue
+            wait_until(lambda: is_queue_created(broker_connection, PIV_QUEUE_NAME))
+
+            mq_handler.publish(str('{"key": "Some valid JSON"}'), PIV_CONTRIBUTOR_ID)
+            wait_until(lambda: RealTimeUpdate.query.count() == 1)
+            # Check lock acquired and released
+            mock_debug.assert_any_call("lock {%s} updated", PIV_LOCK_NAME)
```

Comment:
* Here you only test lock refresh, and it seems a bit lucky that the processing is longer than the interval between refresh, no?
* Can you test lock is acquired through `mock_debug.assert_any_call("getting lock %s", PIV_LOCK_NAME)`?
* It's nice to test that processing is still done, but it's not needed to acquire the lock (so it may be replaced by the sleep only if it's already tested elsewhere)
* No check that it's released (should only be the case if conf changes I guess and doable via `mock_debug.assert_any_call("releasing lock %s", PIV_LOCK_NAME)`).

```suggestion
            # Check lock acquired and refreshed
            mock_debug.assert_any_call("getting lock %s", PIV_LOCK_NAME)
            sleep((PIV_WORKER_REDIS_TIMEOUT_LOCK // 5) + 1)
            mock_debug.assert_any_call("lock {%s} updated", PIV_LOCK_NAME)
```

Reply:
We can't test getting and releasing lock because we only test the worker and not the wrapper

Reply:
Ah, true!
Then only the comment needs a "refresh" :stuck_out_tongue_closed_eyes: